test/libsolidity/hooks/SmokeHook.h

```diff
@@ -0,0 +1,46 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <iostream>
+#include <test/libsolidity/TestHook.h>
+
+namespace solidity::frontend::test
+{
+class SmokeHook: public TestHook
+{
+public:
+	SmokeHook() = default;
+	~SmokeHook() override = default;
+
+	void beginTestCase() override {}
+	void beforeFunctionCall([[maybe_unused]] TestFunctionCall const& _call) override {}
+	void afterFunctionCall([[maybe_unused]] TestFunctionCall const& _call) override {}
+	void endTestCase() override {}
```

Comment:
I am still thinking that all classes that are inheriting from `TestHook` should implement this 4 functions, at least as empty function.
Would really make them abstract in `TestHook` instead of doing default noop implementation there.
Any other opinions?

Reply:
Ok, I think I will now just make the whole class abstract.